tests/std/tests/GH_000685_condition_variable_any/test.cpp

```diff
@@ -79,9 +82,136 @@ namespace {
         assert(m.num_locks() == 4);
 #endif // _HAS_CXX20
     }
+
+    // Minimal example inspired by LWG-4301, modified due to missing latch in C++11 to 20
+    // and generalized to test all overloads of condition_variable{_any}::wait_{for, until}
+    // Idea: Make the main thread wait for a CV with a short timeout and modify it from another thread in the meantime.
+    // If the main thread wait times out after short time, the modification did not influence the ongoing wait
+    template <typename CV>
+    void test_timeout_immutable(int test_number, int retries_remaining = 5) {

```

Comment:
We're less disciplined in test code, but in a large function, it's especially nice to make value parameters const when possible.